### DISCUSSION TITLE
Spectral power distribution - column and colorful chart

### DISCUSSION THREAD

**Discussion by latonita** [dbuezas/lovelace-plotly-graph-card#385] [Q&A] [answered]
Hi.
I have spectral sensor which provides 11 points of data (separate entities for each wavelength).
I managed to make a bar chart with 'custom bar char' component. But I'm not totally happy with it :)
![image](https://github.com/dbuezas/lovelace-plotly-graph-card/assets/3993466/4fb1471b-d03c-401e-80f3-1b2792fe380b)

0) Is there an example how to do similar thing with plotly?
1) Is it possible to build some kind of smoothed line with your plotly component? I understand 11 points of data is not enough.
![image](https://github.com/dbuezas/lovelace-plotly-graph-card/assets/3993466/89d8a7c1-2e5e-4811-a9ea-6c8d41c512b5)
2) If 1) is possible, I can make an algorithm to calculate all the points from 350nm to 1000 nm. Would it be possible to run this js algorithm and visualize smoothed line chart ?

Thank you

**Comment by dbuezas** [comment]
Plotly unfortunately doesn't support gradient filling, so the coloring won't work without dynamically generating traces. And that's not supported in this card (it was hard to do without affecting performance)

The only options I see are:
* Drop the colors and use a line scatter with filling
* Use shapes instead of traces (see plotly docs)
* make a program to generate a huge yaml and paste the result as the card config


Good job by the way, interesting plot!
What do you use the spectrum for?

**Reply by latonita** [reply]
Okay, thanks this makes sense. An i think scatter with filling might work.

Is there an example how to make simple column chart? I'm still struggling with how to make X axis not the historical data, but latest data from different sensors
Thank you

**Reply by latonita** [reply]
What is it for: trying to use spectral sensor for gardening:  calculating irradiance, trying to calculate PPFD and trying to show spectrum. So this is to test lightning in different parts of greenhouse and have basic info regarding lightning. Can be used to integrate overall energy received by plants.

**Reply by latonita** [reply]
Can be useful in lightning automation later (when integrating energy through the day)

**Reply by dbuezas** [reply, marked as answer]
Wow, that's really cool!

> how to make X axis not the historical data, but latest data from different sensors

See this example: https://github.com/dbuezas/lovelace-plotly-graph-card/discussions/298
It's for 3d scatters, but you see there how to define all axes.

**Reply by latonita** [reply]
Thanks, this is nice example!
